controllers/storagecluster/reconcile.go

```diff
@@ -212,6 +212,12 @@ func (r *StorageClusterReconciler) validateStorageClusterSpec(instance *ocsv1.St
 		r.Log.Error(err, "Failed to validate StorageCluster version.", "StorageCluster", klog.KRef(instance.Namespace, instance.Name))
 		r.recorder.ReportIfNotPresent(instance, corev1.EventTypeWarning, statusutil.EventReasonValidationFailed, err.Error())
 		instance.Status.Phase = statusutil.PhaseError
+		instance.Status.Conditions = append(instance.Status.Conditions, conditionsv1.Condition{
```

Comment:
Actually, you cannot append like that. It is possible that 'conditionsv1.ConditionDegraded' is already a part of the slice, in which case we will have two identical conditions and many more with the same thing because it will be appended on every reconcile.

Reply:
Got it.
Just noticed that we have a utility function `setStatusConditionIfNotPresent` which would handle that, though it was private for some reason. So, I have exported it and used that function to handle the append of status. I think now it should be fine.